### ISSUE TITLE
[Bug] about training time

### ISSUE THREAD

**Issue by mc-lan** [open-mmlab/mmselfsup#619]
### Branch

master branch (0.x version, such as `v0.10.0`, or `dev` branch)

### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmselfsup/issues) and [Discussions](https://github.com/open-mmlab/mmselfsup/discussions) but cannot get the expected help.
- [X] I have read the [documentation](https://mmselfsup.readthedocs.io/en/latest/) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmselfsup).

### Environment

fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
sys.platform: linux
Python: 3.9.13 (main, Aug 25 2022, 23:26:10) [GCC 11.2.0]
CUDA available: True
GPU 0,1,2,3: NVIDIA GeForce RTX 3090
CUDA_HOME: None
GCC: gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
PyTorch: 1.12.1
PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.6
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.3.2  (built against CUDA 11.5)
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.6, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.13.1
OpenCV: 4.6.0
MMCV: 1.6.2
MMCV Compiler: GCC 9.3
MMCV CUDA Compiler: 11.6
MMSelfSup: 0.10.1+

### Describe the bug

It seems that the training time of using 1 3090 equals to the time of using 4 3090s. The experiment is conducted on a server with 4 3090s. I wonder if this is reasonable？

### Reproduces the problem - command or script

# one 3090
bash tools/dist_train.sh configs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224.py \
1 \
--work_dir work_dirs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224/

# four 3090s

bash tools/dist_train.sh configs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224.py \
4 \
--work_dir work_dirs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224/

### Reproduces the problem - error message

# message of using one 3090
2022-12-08 05:15:05,285 - mmselfsup - INFO - Checkpoints will be saved to /data/mclan/mmselfsup/work_dirs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224 by HardDiskBackend.
2022-12-08 05:15:17,654 - mmcv - INFO - Reducer buckets have been rebuilt in this iteration.
2022-12-08 05:16:28,176 - mmselfsup - INFO - Epoch [1][50/10010]        lr: 5.337e-07, eta: 57 days, 14:49:58, time: 1.658, data_time: 1.143, memory: 18113, loss: 3.8216
2022-12-08 05:17:42,463 - mmselfsup - INFO - Epoch [1][100/10010]       lr: 8.333e-07, eta: 54 days, 15:03:26, time: 1.486, data_time: 1.018, memory: 18113, loss: 3.7076
2022-12-08 05:18:58,989 - mmselfsup - INFO - Epoch [1][150/10010]       lr: 1.133e-06, eta: 54 days, 3:34:22, time: 1.531, data_time: 1.066, memory: 18113, loss: 3.6088
2022-12-08 05:20:14,082 - mmselfsup - INFO - Epoch [1][200/10010]       lr: 1.433e-06, eta: 53 days, 15:50:29, time: 1.502, data_time: 1.038, memory: 18113, loss: 3.5394
2022-12-08 05:21:30,916 - mmselfsup - INFO - Epoch [1][250/10010]       lr: 1.732e-06, eta: 53 days, 14:36:14, time: 1.537, data_time: 1.072, memory: 18113, loss: 3.4960
2022-12-08 05:22:44,690 - mmselfsup - INFO - Epoch [1][300/10010]       lr: 2.032e-06, eta: 53 days, 5:15:51, time: 1.475, data_time: 1.011, memory: 18114, loss: 3.4509
2022-12-08 05:24:02,147 - mmselfsup - INFO - Epoch [1][350/10010]       lr: 2.332e-06, eta: 53 days, 7:21:48, time: 1.549, data_time: 1.083, memory: 18114, loss: 3.4217
2022-12-08 05:25:20,036 - mmselfsup - INFO - Epoch [1][400/10010]       lr: 2.631e-06, eta: 53 days, 9:49:56, time: 1.558, data_time: 1.090, memory: 18114, loss: 3.3787
2022-12-08 05:26:35,585 - mmselfsup - INFO - Epoch [1][450/10010]       lr: 2.931e-06, eta: 53 days, 7:24:40, time: 1.511, data_time: 1.043, memory: 18114, loss: 3.3571

# message of using 4 3090
2022-12-08 05:27:41,076 - mmselfsup - INFO - workflow: [('train', 1)], max: 300 epochs
2022-12-08 05:27:41,078 - mmselfsup - INFO - Checkpoints will be saved to /data/mclan/mmselfsup/work_dirs/selfsup/mocov3/mocov3_vit-small-p16_32xb128-fp16-coslr-300e_in1k-224 by HardDiskBackend.
2022-12-08 05:28:17,034 - mmcv - INFO - Reducer buckets have been rebuilt in this iteration.
2022-12-08 05:33:22,157 - mmselfsup - INFO - Epoch [1][50/2503] lr: 1.414e-06, eta: 59 days, 6:45:03, time: 6.821, data_time: 3.468, memory: 18112, loss: 4.8723
2022-12-08 05:38:30,091 - mmselfsup - INFO - Epoch [1][100/2503]        lr: 2.613e-06, eta: 56 days, 9:32:46, time: 6.159, data_time: 1.307, memory: 18112, loss: 4.6634
2022-12-08 05:43:49,527 - mmselfsup - INFO - Epoch [1][150/2503]        lr: 3.811e-06, eta: 56 days, 2:24:33, time: 6.389, data_time: 0.182, memory: 18112, loss: 4.5525
2022-12-08 05:49:00,344 - mmselfsup - INFO - Epoch [1][200/2503]        lr: 5.010e-06, eta: 55 days, 13:48:31, time: 6.216, data_time: 0.000, memory: 18112, loss: 4.4745
2022-12-08 05:54:20,376 - mmselfsup - INFO - Epoch [1][250/2503]        lr: 6.208e-06, eta: 55 days, 13:53:58, time: 6.401, data_time: 0.000, memory: 18112, loss: 4.4100

**Comment by YuanLiuuuuuu** [comment]
I think there are problems with your local machine. Just as your log shown, the time of 50 iters for 4 3090s is approximately 4 times of that for 1 3090, which accounts for the equal pre-training time.

**Comment by mc-lan** [comment]
> 



> I think there are problems with your local machine. Just as your log shown, the time of 50 iters for 4 3090s is approximately 4 times of that for 1 3090, which accounts for the equal pre-training time.

thank you, will make a check.

**Comment by YuanLiuuuuuu** [comment]
Close with being solved, if you have any other problems, feel free to reopen it. Thanks!

**Comment by Lan-MC** [comment]
Kind reminder: it was mp_start_method = 'fork' that leads to super slow speed of dataloder because of deadlock. If someone encounter the same problem, please reset it to mp.set_start_method('spawn').

**Comment by mc-lan** [comment]
Kind reminder: it was mp_start_method = 'fork' that leads to a super slow speed of dataloder because of the deadlock. If someone encounter the same problem, please reset it to mp.set_start_method('spawn').
